src/runner/browser-env/vite/plugins/generate-index-html.ts

```diff
@@ -46,6 +51,15 @@ export const plugin = async (): Promise<Plugin[]> => {
 
     const automationProtocolPath = `/@fs${driverModulePath}`;
 
+    const mockDefaultModulePath = path.resolve(browserModulesPath, "mock/default-module.js");
+    const mockImportMetaResolvePath = path.resolve(browserModulesPath, "mock/import-meta-resolve.js");
+    const mockWdioLoggerPath = path.resolve(browserModulesPath, "mock/@wdio-logger.js");
+
+    const modulesToMock = DEFAULT_MODULES_TO_MOCK.reduce((acc, val) => _.set(acc, val, mockDefaultModulePath), {
+        "@wdio/logger": mockWdioLoggerPath,
+        "import-meta-resolve": mockImportMetaResolvePath,
+    }) as Record<string, string>;
```

Comment:
here, instead of using virtual modules, I started using mock files from the file system